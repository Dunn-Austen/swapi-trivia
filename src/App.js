import React, { Component } from 'react';
import './App.css';
import MovieContainer from './MovieContainer';
import Login from './Login';


class App extends Component {
  constructor() {
    super();
    this.state = {
      user: {
        name: '',
        quote: '',
        ranking: ''
      }
    }
  }

  enterUserInfo = (userData) => {
    this.setState({
      user:{...userData}
    })
  }

  render() {
    return (
      <main>
        <div>
          <Login enterUserInfo = {this.enterUserInfo}/>
          {/* <MovieContainer foster={['yello', 'dave', 'its', 'all', 'coming', 'together']}/> */}
          </div>
      </main>
    )
  }
}

export default App;
