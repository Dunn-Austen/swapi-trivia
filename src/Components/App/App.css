* {
  margin: 0;
}

main {
  background: linear-gradient(rgba(0, 0, .5, .2), rgba(0, .5, .5, 1)), url("../../Images/Death-star2.jpg");
  background-blend-mode: saturation;
  background-repeat: no-repeat;
  background-size: 100% 100%;
  height: 100vh;
  position: relative;
  text-align: center;
}

.userdata-component {
  position: absolute;
  top: 0;
}

.movie-container {
  position: relative;
  top: 20%;
}
