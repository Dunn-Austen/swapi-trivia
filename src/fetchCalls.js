export const retrieveAllMovies = (url) => {
    return fetch(url)
        .then(response => response.json())
        .then(movies => movies.results)
        .catch(error => console.log(error))
}



export const retrieveAllCharacters = (characters) => {
    const characterInfo = characters.map(character => {
        return getCharacterData(character)
            .then(character => ({
                name: character[0],
                homeWorld: character[1].home,
                population: character[1].population,
                species: character[2].species,
                relatedFilms:character[3]
            }))
    })
    return Promise.all(characterInfo)
}
    
const getCharacterData = (url) => {
    return fetch(url)
        .then(response => response.json())
        .then(character => {
            const { name } = character;
            const homeWorld = getHomeWorld(character.homeworld);
            const species = getSpecies(character.species);
            const films = getrelatedFilms(character.films)
                return Promise.all([name, homeWorld, species, films])
        })
}

const getHomeWorld = (homeworldUrl) => {
    return fetch(homeworldUrl)
        .then(response => response.json())
        .then(data => {
            return ({
                home: data.name,
                population: data.population
            })
        })
}
    

const getSpecies = (speciesUrl) => {
    return fetch(speciesUrl)
        .then(response => response.json())
        .then(data =>   {
            return ({
                species: data.name,
            })
        })
}

const getrelatedFilms = (films) => {
    const filmData = films.map(film => {
        return getFilmName(film)
        .then(film => film)
    })
    return Promise.all(filmData)
}

const getFilmName = (url) => {
    return fetch(url)
        .then(response => response.json())
        .then(data => data.title)
}