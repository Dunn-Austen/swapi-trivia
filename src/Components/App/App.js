import React, { Component } from 'react';
import './App.css';
import MovieContainer from '../MovieContainer/MovieContainer';
import Login from '../Login/Login';
import UserData from '../UserData/UserData';


class App extends Component {
  constructor() {
    super();
    this.state = {
      user: {
        name: '',
        quote: '',
        ranking: '',
      },
    }
  }

  enterUserInfo = (userData) => {
    this.setState({
      user: {...userData}
    })
  }

  logOut = () => {
    this.setState({
      user: {
        name: '',
        quote: '',
        ranking: '',
        isSignedIn: false,
        errors: {}
      },
    })
  }


  render() {
    return (
      <main>
        <div>
          <Login enterUserInfo = {this.enterUserInfo}/>
          {/* <MovieContainer foster={['yello', 'dave', 'its', 'all', 'coming', 'together']}/> */}
          <UserData logOut={this.logOut} name={this.state.user.name} quote={this.state.user.quote} ranking={this.state.user.ranking}/>
        </div>
      </main>
    )
  }
}

export default App;
