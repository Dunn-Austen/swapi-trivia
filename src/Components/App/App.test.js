import React from 'react';
import { App } from './App';
import { shallow } from 'enzyme';
import { retrieveAllMovies } from '../../fetchCalls';

jest.mock('../../fetchCalls.js')

describe('App', () => {
  beforeEach(() => {
    retrieveAllMovies.mockImplementation(() => {
      return Promise.resolve([{movie1: 'name'}, {movie2: 'name'}, {movie3: 'name'}])
    });
  });

    it('should match the snapshot', () => {
      const wrapper = shallow(<App />);
      expect(wrapper).toMatchSnapshot()
    });

    it('should invoke retrieveAllMovies after componentDidMount triggers', () => {
      shallow(<App />);
      expect(retrieveAllMovies).toHaveBeenCalled();
    });
      //Imports are solid, syntax is fine, but this still invoke in the test (0 invokes)?
      //Shallow should trigger componentDidMount which invokes retrieveAllMovies
      //I've tried declaring const retrieveAllMovies = jest.fn.mockimpl() as an alternative with no dice
      //File pathway isn't the problem
      //I've tried writing the jest.mock(url with and without the .js suffix)
      //Jest dependency IS installed; the snapshot test works
      //I've tried breaking the test into a nested described('componentDidMount' ...), but that didn't work
      //I tried to first manually invoke componentDidMount (wrapper.instance().componentDidMount();)
      //Tried playing with async / await in strange ways
      //Could it have something to do with our class syntax in App.js (App extends React.Component instead of Component)?


    it('should update state when enterUserInfo is called', () => {
      const wrapper = shallow(<App />);
      const mockUserData = {name: 'Billy Madison', quote: 'Odoyle juuls', ranking: 'Padawan'};
      const expected = {name: 'Billy Madison', quote: 'Odoyle juuls', ranking: 'Padawan'};
        // wrapper.instance().enterUserInfo = jest.fn();
        // wrapper.instance().forceUpdate();
      expect(wrapper.state('user')).toEqual({});

      wrapper.instance().enterUserInfo(mockUserData);

      expect(wrapper.state('user')).toEqual(expected);
    });
    //Test won't register App as a class component (error message: ShallowWrapper::state() can only be called on class components)
    //Tried various unsuccessful tweaks - is there some global issue with the App file? Import syntax?
    //SOLVED - import syntax

    it('should update state when logOut is called', () => {
      const wrapper = shallow(<App />);
      const mockUserData = {name: 'Billy Madison', quote: 'Odoyle juuls', ranking: 'Padawan'};
      // wrapper.instance().enterUserInfo = jest.fn();
      // wrapper.instance().forceUpdate();
      // expect(wrapper.state('user')).toEqual(mockUserData);
      wrapper.instance().setState({user: mockUserData});

      wrapper.instance().logOut();

      expect(wrapper.state()).toEqual(
                  {
                    favoritedCharacters: [],
                    favoritesCounter: 0,
                    flicks: [],
                    openingCrawl: null,
                    selectedCharacters: [],
                    selectedFlick: {},
                    user: {
                      name: "",
                      quote: "",
                      ranking: "",
                    }
                  });
    });



  //
  //   logOut = () => {
  //   {/*   this.props.history.push('/')   */}
  //     this.setState({
  //       user: {
  //         name: '',
  //         quote: '',
  //         ranking: ''
  //       },
  //     })
  //   }
  //
  //
  //
  //
  // selectFlick = (id) => {
  //   let correctCrawl = this.state.flicks.find(flick => {
  //     return flick.episode_id === parseInt(id)
  //   })
  //   {/*  this.props.history.push(`/movies:${id}`)   */}
  //   this.setState({ selectedFlick: this.state.flicks[id - 1], openingCrawl: correctCrawl.opening_crawl})
  //   this.findCharacters(this.state.flicks[id - 1].characters)
  // }




});
