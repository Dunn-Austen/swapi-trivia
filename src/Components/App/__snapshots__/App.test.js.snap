// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
<main>
  <Route
    exact={true}
    path="/"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/movies"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/movies"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/movies:id"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/movies/favorites"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/movies:id"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/movies:id"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/movies/favorites"
    render={[Function]}
  />
</main>
`;
